src/legacy/ui/public/field_editor/components/field_format_editor/editors/currency/currency.js

```diff
@@ -0,0 +1,368 @@
+    code: 'DKK',
+    symbol: 'kr',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.PLN', {
+      defaultMessage: 'Polish zloty',
+    }),
+    code: 'PLN',
+    symbol: 'zł',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.THB', {
+      defaultMessage: 'Thai baht',
+    }),
+    code: 'THB',
+    symbol: '฿',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.IDR', {
+      defaultMessage: 'Indonesian rupiah',
+    }),
+    code: 'IDR',
+    symbol: 'Rp',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.HUF', {
+      defaultMessage: 'Hungarian forint',
+    }),
+    code: 'HUF',
+    symbol: 'Ft',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.CZK', {
+      defaultMessage: 'Czech koruna',
+    }),
+    code: 'CZK',
+    symbol: 'Kč',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.ILS', {
+      defaultMessage: 'Israeli new shekel',
+    }),
+    code: 'ILS',
+    symbol: '₪',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.CLP', {
+      defaultMessage: 'Chilean peso',
+    }),
+    code: 'CLP',
+    symbol: 'CLP$',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.PHP', {
+      defaultMessage: 'Philippine peso',
+    }),
+    code: 'PHP',
+    symbol: '₱',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.AED', {
+      defaultMessage: 'UAE dirham',
+    }),
+    code: 'AED',
+    symbol: 'د.إ',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.COP', {
+      defaultMessage: 'Colombian peso',
+    }),
+    code: 'COP',
+    symbol: 'COL$',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.SAR', {
+      defaultMessage: 'Saudi riyal',
+    }),
+    code: 'SAR',
+    symbol: '﷼',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.MYR', {
+      defaultMessage: 'Malaysian ringgit',
+    }),
+    code: 'MYR',
+    symbol: 'RM',
+  },
+  {
+    name: i18n.translate('common.ui.fieldEditor.currency.currencies.RON', {
+      defaultMessage: 'Romanian leu',
+    }),
+    code: 'RON',
+    symbol: 'L',
+  },
+];
+
+export class CurrencyFormatEditor extends DefaultNumberFormatEditor {
+  static formatId = 'currency';
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      ...this.state,
+      sampleInputs: [1234, 99.9999, 5150000.0001, 0.00005],
+      hasOther: false,
+    };
+  }
+
+  render() {
+    const { formatParams } = this.props;
+    const { samples } = this.state;
+    const currencyCode = formatParams.currencyCode;
+    const currencyMatch = topCurrencies.find(cur => cur.code === currencyCode);
+    let currencyLabel = currencyCode;
+    if (currencyMatch) {
+      currencyLabel = `${currencyMatch.name} (${currencyMatch.code}) ${currencyMatch.symbol}`;
```

Comment:
:art: Since we use this way of formatting here and in line 332 we could potentially create a very small `formatCurrencyName` function for this in this file, so we don't accidentally later on break alignment between those two formatting.